api/app.go

```diff
@@ -215,11 +215,24 @@ func appList(w http.ResponseWriter, r *http.Request, t auth.Token) error {
 		w.WriteHeader(http.StatusNoContent)
 		return nil
 	}
+	w.Header().Set("Content-Type", "application/json")
+	simple, _ := strconv.ParseBool(r.URL.Query().Get("nameOnly"))
+	if simple {
+		appsNames := make([]interface{}, len(apps))
```

Comment:
Instead of using a `[]interface{}`, I think it is better to use a `[]miniApp` here and then set each miniApp's name on the loop.

Reply:
I agree. In fact, that was my first attempt. However, since `miniApp` struct fields implement json decoder without  `omitempty`, I ended with the following JSON which I think it is not correct since it implies the App has all its fields empty.

```
[{
    "name": "App1",
    "pool": "",
    "teamowner": "",
    "plan": {
        "name": "",
        "memory": 0,
        "swap": 0,
        "cpushare": 0
    },
    "units": null,
    "cname": null,
    "ip": "",
    "routers": null,
    "lock": null,
    "tags": null
}]
```


In order to try and test another attempt, I added `omitempty`to `miniApp`  and ended with a result much closer to the expected solution: 

```
[{"name":"App1","plan":{"name":"","memory":0,"swap":0,"cpushare":0}}]
```

However, I feel this is not quite right since I had to modify the whole miniApp json struct just for this issue. That is why I went with `[]interface{}`

I looked around but almost every solution used reflection to remove the empty struct fields on the fly which I think its an overhead... What do you think?

Reply:
Hi @juanjcsr, thanks for getting back. I was thinking about this issue one more time and I think we could simplify (pun not intended) our approach:

Instead of using `nameOnly` as a query flag, we could use `simplified` then simply return the entire `miniApp` without the units (which is the complex part). You could do that just by skipping the call to `app.Units` if this flag is set.

What do you think?

Reply:
I think this is a way better solution. It solves the original issue while still being fast by skipping the Units part. 

I will upload my changes latter today and will also add documentation to it.

Reply:
Hi @andrestc, I updated my code to support a new `simplified` flag that returns an array of `miniApp` without the `app.Units` part.